File: 2860-happy-students/2860-happy-students.java
class Solution {
        public int countWays(List<Integer> nums) {
        int[] arr = new int[nums.size()];
        for(int i = 0;i<nums.size();i++){
            arr[i] = nums.get(i);
        }
        Arrays.sort(arr);
        int ans = 0;
        // smaller than group | larger than group
        int numSelected = 0;

        // {}| larger than group
        if(arr[0] > numSelected){
            ans++;
        }
        for(int i = 0;i<arr.length;i++){
            //select i-th student;
            //{nums.get(0),..., nums.get(i)} | {nums.get(i+1),...}
            numSelected++;
            if(numSelected > arr[i] 
                && (i == arr.length - 1 || numSelected < arr[i+1] )) {
                ans++;
            }
        }
        return ans;
    }
}

File: 2860-happy-students/2860-happy-students.cpp
class Solution {
public:
    int countWays(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int ans = 0, n = nums.size();
        int selected = 0;
        if(nums[0]!=0) ans = 1; // Not Selecting AnyOne
        for(int i=0;i<n;i++) {
            selected++;
            if(selected>nums[i]) {  // No. of Selected Students is strictly greater than nums[i].
                if(i+1<n && selected<nums[i+1]) // Considering from (i+1) to n Students is not Selected.
                    ans++; 
                else if(i+1==n) ans++;  // Last Student
            }
        }
        return ans;
    }
};
